pkg/component/runtime/command.go

```diff
@@ -69,6 +80,7 @@ func NewCommandRuntime(comp component.Component) (ComponentRuntime, error) {
 // ever be called again.
 func (c *CommandRuntime) Run(ctx context.Context, comm Communicator) error {
 	checkinPeriod := c.current.Spec.Spec.Command.Timeouts.Checkin
+	restartPeriod := c.current.Spec.Spec.Command.Timeouts.Restart
```

Comment:
can we run into nil pointer somewhere here?

Reply:
This is checked in the initialization of the struct. The depth after that does not have any pointers.

```
if comp.Spec.Spec.Command == nil {
		return nil, errors.New("must have command defined in specification")
	}
```